index.d.ts

```diff
@@ -19,6 +29,117 @@ export interface MonacoTailwindcssOptions {
    * worker.
    */
   tailwindConfig?: TailwindConfig | string
+  /**
+   * Extend Intellisense's settings.
+   * 
+   * Default values are based on the VS Code extension.
+   * 
+   * ``` typescript
+   * const defaultSettings = {
+      editor: {tabSize: 2},
+      tailwindCSS: {
+        emmetCompletions: false,
+        classAttributes: ["class", "className", "ngClass"],
+        codeActions: true,
+        hovers: true,
+        suggestions: true,
+        validate: true,
+        colorDecorators: true,
+        rootFontSize: 16,
+        showPixelEquivalents: true,
+        includeLanguages: {},
+        files: {exclude: []},
+        experimental: {
+          classRegex: [],
+          configFile: {}
+        },
+        lint: {
+          cssConflict: "warning",
+          invalidApply: "error",
+          invalidScreen: "error",
+          invalidVariant: "error",
+          invalidConfigPath: "error",
+          invalidTailwindDirective: "error",
+          recommendedVariantOrder: "warning"
+        },
+      }
+    }
+   */
+  intellisense?: Partial<{
+    editor: {
+      tabSize: number
+    }
+    tailwindCSS: Partial<{
+      emmetCompletions: boolean
+      includeLanguages: Record<string, string>
+      classAttributes: string[]
+      suggestions: boolean
+      hovers: boolean
+      codeActions: boolean
+      validate: boolean
+      showPixelEquivalents: boolean
+      rootFontSize: number
+      colorDecorators: boolean
+      files: { exclude: string[] }
+      experimental: {
+        classRegex: string[]
+        configFile: string | Record<string, string | string[]>
+      }
+      lint: {
+        cssConflict: DiagnosticSeveritySetting
+        invalidApply: DiagnosticSeveritySetting
+        invalidScreen: DiagnosticSeveritySetting
+        invalidVariant: DiagnosticSeveritySetting
+        invalidConfigPath: DiagnosticSeveritySetting
+        invalidTailwindDirective: DiagnosticSeveritySetting
+        recommendedVariantOrder: DiagnosticSeveritySetting
+      }
+    }>
+  }>
```

Comment:
I think all of these nested properties should be optional, also the deeply nested ones. Since we’re defining the object, use the optional property symbol `?`, not the `Partial` type.

Reply:
Noted